pyccel/codegen/printing/ccode.py

```diff
@@ -853,11 +931,34 @@ def _print_AugAssign(self, expr):
         return "{0} {1}= {2};".format(lhs_code, op, rhs_code)
 
     def _print_Assign(self, expr):
+        lhs = self._print(expr.lhs)
+        rhs = expr.rhs
+        if isinstance(rhs, (NumpyArray)):
+            dummy_array_name = "dummy_%s" % (rhs.__hash__()%991)
+            dtype = self.find_in_dtype_registry(format(rhs.dtype), rhs.precision)
+            arg = rhs.arg
+            if rhs.rank > 1:
+                import functools
+                import operator
+                arg = functools.reduce(operator.concat, arg)
```

Comment:
What are you doing here?

Reply:
this is to flatten the array element into one list